test/Effect/applicative.ts

```diff
@@ -0,0 +1,16 @@
+import * as it from "effect-test/utils/extend"
+import * as Effect from "effect/Effect"
+import * as Fiber from "effect/Fiber"
+import * as Queue from "effect/Queue"
+import { assert, describe } from "vitest"
+
+describe.concurrent("Queue", () => {
+  it.effect("applicative take and offer", () =>
+    Effect.gen(function*($) {
+      const queue = yield* $(Queue.bounded<string>(100))
+      const fiber = yield* $(Queue.take(queue), Effect.ap(Queue.take(queue), (a, b) => a + b), Effect.fork)
```

Comment:
@gcanti the test works but TS blows in this line, I guess I need to type cast to something more specific? 🤔

Reply:
Not sure, but honestly l would switch to a simpler test using `Effect.succeed` and `Effect.fail`, along the lines of what we did with `Either`